lib/awful/rules.lua

```diff
@@ -532,12 +674,11 @@ function rules.execute(c, props, callbacks)
     end
 end
 
+-- TODO v5 deprecate this
 function rules.completed_with_payload_callback(c, props, callbacks)
     rules.execute(c, props, callbacks)
 end
 
-client.connect_signal("spawn::completed_with_payload", rules.completed_with_payload_callback)
```

Comment:
Can you also add a deprecated-todo to the corresponding bits in `awful.spawn` (as a separate commit)?